### ISSUE TITLE
Slices of structs are not properly reset when unmarshalling

### ISSUE THREAD

**Issue by kkoreilly** [pelletier/go-toml#931]
**Describe the bug**
When unmarshalling a slice of structs (array of tables in TOML terminology) into a Go slice, any elements already present in the slice are not reset, leading to compounding slices. This did not happen in v1.

**To Reproduce**

`main.go`:
```go
package main

import (
	"fmt"
	"os"

	"github.com/pelletier/go-toml/v2"
)

type item struct {
	Name string
}

type items struct {
	Slice []item
}

func main() {
	its := items{[]item{{"a"}, {"b"}}}
	b, _ := os.ReadFile("slice.toml")
	toml.Unmarshal(b, &its)
	fmt.Println(its) // should be {[{c} {d}]}, but get {[{a} {b} {c} {d}]}
}
```

`slice.toml`:
```toml
[[Slice]]
  Name = 'c'

[[Slice]]
  Name = 'd'
```

**Expected behavior**
As stated in the code, I expected `{[{c} {d}]}` but got `{[{a} {b} {c} {d}]}`.

**Versions**
 - go-toml: v2.1.1 (34765b4)
 - go: 1.21.4
 - operating system: Windows and macOS

**Additional context**
As stated above, I have confirmed that this issue did not happen in v1.

**Comment by pelletier** [comment]
Good catch, thank you for the bug report!

The intent is to match `encoding/json`'s behavior when unmarshaling. I've confirmed (https://go.dev/play/p/WQ8VPf5C-L1) it resets the slice and the resulting value is `{[{c} {d}]}`.

**Comment by pelletier** [comment]
@kkoreilly I've merged #934, which should address your issue. Let me know if it worked, feel free to re-open this issue if not!

**Comment by kkoreilly** [comment]
It worked, thank you!
